kong/runloop/balancer.lua

```diff
@@ -656,6 +657,20 @@ local create_hash = function(upstream)
       if type(identifier) == "table" then
         identifier = table_concat(identifier)
       end
+
+    elseif hash_on == "cookie" then
+      identifier = ngx.var["cookie_" .. upstream.hash_on_cookie]
+
+      -- If the cookie doesn't exist, create it
+      if not identifier then
+        identifier = utils.uuid()
+        ngx.ctx.balancer_hash_cookie = {
```

Comment:
`ctx` is already locally cached above in this function, so better use it instead of invoking `ngx.ctx` again here.